server/src/main/java/org/apache/druid/segment/metadata/CoordinatorSegmentMetadataCache.java

```diff
@@ -826,22 +824,20 @@ Optional<RowSignature> mergeOrCreateRowSignature(
 
       for (String column : segmentSchema.getNewColumns()) {
         if (mergedColumnTypes.containsKey(column)) {
-          existingNewColumns = true;
           mergedColumnTypes.compute(column, (c, existingType) -> columnTypeMergePolicy.merge(existingType, columnMapping.get(column)));
         } else {
           mergedColumnTypes.put(column, columnMapping.get(column));
         }
       }
 
-      if (missingUpdateColumns || existingNewColumns) {
+      if (!missingUpdateColumns.isEmpty()) {
         log.makeAlert(
-            "Error merging delta schema update with existing row signature. segmentId [%s], "
-            + "existingSignature [%s], deltaSchema [%s], missingUpdateColumns [%s], existingNewColumns [%s].",
-            segmentId,
+            "Realtime schema merge: datasource [%s] is missing columns to be updated. "
+            + "ExistingSignature [%s], deltaSchema [%s], missingUpdateColumns [%s].",
```

Comment:
The alert description should be small. Don't include big objects like the `existingSignature`or `deltaSchema` in it. You may add these fields separately using `.addData()`.